src/win/util.c

```diff
@@ -1676,14 +1676,27 @@ int uv_os_gethostname(char* buffer, size_t* size) {
     return uv_translate_sys_error(WSAGetLastError());
 
   buf[sizeof(buf) - 1] = '\0'; /* Null terminate, just to be safe. */
-  len = strlen(buf);
+  int utf16StrBufSize = MultiByteToWideChar(CP_ACP, 0, buf, -1, NULL, 0);
+  WCHAR* utf16Str = uv__malloc(sizeof(WCHAR) * utf16StrBufSize);
+  MultiByteToWideChar(CP_ACP, 0, buf, -1, utf16Str, utf16StrBufSize);
 
+  char* utf8Str = uv__malloc(sizeof(char) * UV_MAXHOSTNAMESIZE);
```

Comment:
I'd use a stack-allocated buffer for this: `char utf8_str[UV_MAXHOSTNAMESIZE];`

Reply:
Yes, I'd like to. But it cause grabled code again

Reply:
> I'd use a stack-allocated buffer for this: `char utf8_str[UV_MAXHOSTNAMESIZE];`

Yes, I'd like too. But it cause grabled code again

Reply:
I'm afraid I don't understand what you mean. What does the change you made look like and what's the output you get afterwards?

Reply:
@bnoordhuis forget my last reply, please :) 
I misunderstood the function of "uv__convert_utf16_to_utf8". Now, I update the code, please review again.